### ISSUE TITLE
Python module synced from PyPI is not found and cannot be installed if it has "." in name

### ISSUE THREAD

**Issue by thomasswainnomad** [pulp/pulp_python#467]
Running the commands to create a PyPI repository, create a remote (including the module "jaraco.collections"), sync the remote, and create a distribution all complete without issue. However, there is some issue when the module name contains a full stop ("."). 

```
pulp python repository create --name test_repo2
pulp python remote create --name test_remote2 --includes '["jaraco-collections"]' --url https://pypi.org/
pulp python repository sync --name test_repo2 --remote test_remote2
pulp python distribution create --name test_dist2 --base-path test_dist2 --repository test_repo2
```

When attempting to install "jaraco.collections", it is not found in the distribution and the install fails. Our pulp server is named "pulp-server".

```
(venv) [tom@tom-pc /tmp/venv $ pip install jaraco.collections --index-url https://pulp-server/pypi/test_dist2/simple/ --trusted-host pulp-server
Looking in indexes: https://pulp-server/pypi/test_dist2/simple/
ERROR: Could not find a version that satisfies the requirement jaraco.collections (from versions: none)
ERROR: No matching distribution found for jaraco.collections
(venv) [tom@tom-pc /tmp/venv $ pip install jaraco-collections --index-url https://pulp-server/pypi/test_dist2/simple/ --trusted-host pulp-server
Looking in indexes: https://pulp-server/pypi/test_dist2/simple/
ERROR: Could not find a version that satisfies the requirement jaraco-collections (from versions: none)
ERROR: No matching distribution found for jaraco-collections
(venv) [tom@tom-pc /tmp/venv $ 
```

Examining the distribution on our pulp server, strange things can be seen regarding the module synced. It is confirmed that "jaraco.collections" has been included. 

![dist_summary](https://user-images.githubusercontent.com/94833108/142841499-4474f42a-4376-4efc-b621-ee109f34ef12.png)

Clicking the "jaraco.collections" link (NOTE: dot "." in the name) from this page, we are taken to a page that should have references to the actual module files, but it is empty, and the server displays the name of the module as "jaraco-collections" (NOTE: dash "-" in the name). 

![click_link_dot](https://user-images.githubusercontent.com/94833108/142841523-e118e128-abf8-4b4b-b6b8-758e616ddfbb.png)

Tweaking the URL from "jaraco-collections" -> "jaraco.collections", we are taken to a page that does successfully show the module files.

![tweak_url_dot](https://user-images.githubusercontent.com/94833108/142841549-ee736f3e-23cc-4daa-bd2e-b25fde59742d.png)

I am convinced therefore that the "jaraco.collections" module is available on our pulp server for install, but the server has confusion about how it handles the "-" vs "." in the module name.

**Comment by gerrod3** [comment]
Good catch! This occurs due to normalization of names from [PEP 503](https://www.python.org/dev/peps/pep-0503/#normalized-names). Should be a simple fix. 

There is a workaround available using publications. Just run `pulp python publication create --repository test_repo2` and then you should be able to pip install. Without publications the simple html pages are generated live upon each request. Publications pre-generate these files and the distribution will check if the latest repository version has a publication it can redirect to. You can use the `autopublish` feature to automatically make new publications after each new repository version with `pulp python repository update --name test_repo2 --autopublish`. This option isn't so good when you are constantly uploading new packages to the repository, but if are only updating the repository by syncing then it's perfectly fine.

**Comment by thomasswainnomad** [comment]
@gerrod3 thanks so much for the help with this. I confirm the workaround works without issue. Will look out for any fixes also!

**Comment by gerrod3** [comment]
@thomasswainnomad The fix has been merged and will be available in the next release in ~2 weeks. Would you like this fix to be backported o 3.5?

**Comment by thomasswainnomad** [comment]
@gerrod3 no it's ok, we can use the next release. Thanks!
